### ISSUE TITLE
double slash in URL makes trouble

### ISSUE THREAD

**Issue by individual-it** [owncloud/core#30846]
### Steps to reproduce
1. install owncloud
2. make a request with an extra `/` in the path e.g. `curl -u admin:admin http://localhost/ocs//v1.php/cloud/capabilities`

### Expected behaviour
owncloud should ignore that extra slash

### Actual behaviour
empty answer

### Server configuration
**Operating system**:
Ubuntu 17.10
**Web server:**
Apache2
**Database:**
Sqlite
**PHP version:**
7.1
**ownCloud version:** (see ownCloud admin page)
10.1 prealpha
**Updated from an older ownCloud or fresh install:**
fresh
**Where did you install ownCloud from:**
git
#### ownCloud log (data/owncloud.log)
```
{"reqId":"m35gCX069ytbtcuGbHbQ","level":3,"time":"2018-03-21T03:37:39+00:00","remoteAddr":"127.0.0.1","user":"--","app":"PHP","method":"GET","url":"\/ocs\/\/v1.php\/cloud\/capabilities","message":"Exception: The requested uri(\/ocs\/v1.php\/cloud\/capabilities) cannot be processed by the script '\/ocs\/\/v1.php') at \/home\/artur\/www\/owncloud-core\/lib\/private\/AppFramework\/Http\/Request.php#663"}
```

**Comment by ownclouders** [comment]
[GitMate.io](https://gitmate.io) thinks possibly related issues are https://github.com/owncloud/core/issues/17384 (Double slash in URL gives error page), https://github.com/owncloud/core/issues/9881 (Double-slash at URL root breaks browsing into subdirectories), https://github.com/owncloud/core/issues/20922 (Logout URL missing trailing slash), https://github.com/owncloud/core/issues/9064 (owncloudcmd is doubling URL in request), and https://github.com/owncloud/core/issues/13326 (Make "Download from URL" configurable).

**Comment by phil-davis** [comment]
This was noticed during some automated acceptance test work, and would be handy to fixup in general so that "people" can build up URL paths without worrying about which bits have ``/`` already on the end or at the beginning. If the resulting path has "go nowhere" ``//`` then ownCloud still copes.

**Comment by individual-it** [comment]
fixed by #30848
